river/river.go

```diff
@@ -232,16 +232,23 @@ func (r *River) prepareRule() error {
 		}
 	}
 
-	for _, rule := range r.rules {
+	rules := make(map[string]*Rule)
+	for key, rule := range r.rules {
 		if rule.TableInfo, err = r.canal.GetTable(rule.Schema, rule.Table); err != nil {
 			return errors.Trace(err)
 		}
 
 		if len(rule.TableInfo.PKColumns) == 0 {
-			return errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table)
+			if r.c.SkipNonPk == false {
```

Comment:
```
if !r.c.SkipNoPKTable  {
    return xxx
}

log.Errorf()
```

Reply:
I can't understand...  
`if !r.c.SkipNoPKTable {}` alright...
but return xxx...??
and where to write log.Errorf...??

plz give me some detail information!!